cmd/crates/stellar-ledger/src/emulator_tests.rs

```diff
@@ -219,7 +199,7 @@ async fn test_sign_tx_hash_when_hash_signing_is_enabled() {
         let ledger = Arc::clone(&ledger);
         async move { ledger.sign_transaction_hash(path, &test_hash).await }
     });
-    let approve = tokio::task::spawn(approve_tx_hash_signature(ui_host_port));
+    let approve = tokio::task::spawn(approve_tx_hash_signature_2(&ledger_testing));
```

Comment:
When trying to use ledger_testing in approve_tx_hash_signature I am seeing an error that I can't quite seem to solve. I think that the issue is that in testcontainers, once the container goes out of scope it drops the docker reference. And then we can't use it again here. I've played around with using Arcs and RefCells, and may be missing something and would love another set of eyes on this!

Reply:
The errors:
```
error[E0597]: `docker` does not live long enough
   --> cmd/crates/stellar-ledger/src/emulator_tests.rs:176:77
    |
175 |     let docker = clients::Cli::default();
    |         ------ binding `docker` declared here
176 |     let ledger_testing = LedgerTesting::new(apps_dir(), LedgerModel::NanoS, &docker);
    |                                                                             ^^^^^^^ borrowed value does not live long enough
...
202 |     let approve = tokio::task::spawn(approve_tx_hash_signature_2(&ledger_testing));
    |                                      -------------------------------------------- argument requires that `docker` is borrowed for `'static`
...
215 | }
    | - `docker` dropped here while still borrowed

error[E0597]: `ledger_testing` does not live long enough
   --> cmd/crates/stellar-ledger/src/emulator_tests.rs:202:66
    |
176 |     let ledger_testing = LedgerTesting::new(apps_dir(), LedgerModel::NanoS, &docker);
    |         -------------- binding `ledger_testing` declared here
...
202 |     let approve = tokio::task::spawn(approve_tx_hash_signature_2(&ledger_testing));
    |                                      ----------------------------^^^^^^^^^^^^^^^-
    |                                      |                           |
    |                                      |                           borrowed value does not live long enough
    |                                      argument requires that `ledger_testing` is borrowed for `'static`
...
215 | }
    | - `ledger_testing` dropped here while still borrowed
```